### PR TITLE
[SCAL-78325] Handle authentication

### PR DESCRIPTION
**Pull request by ayonghosh**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -13,8 +13,17 @@
...
     None = 'None',
-    SAML = 'SSO_SAML',
+    /**
+     * SSO using SAML
+     */
+    SSO = 'SSO_SAML',
+    /**
+     * Truted authentication server
+     */
     AuthServer = 'AuthServer',
```
Comment (ashubham) [inline src/types.ts:27]: We could also support basic username password auth here.

That was a requirement from Dave, for customers who just want to start development without getting into PROD level auth mechanisms.

File: src/types.ts [hunk 2]
```diff
@@ -27,9 +36,35 @@ export type DOMSelector = string | HTMLElement;
...
+     * and returns a Promise that resolves to the auth token.
+     * To perform trusted auth one of authEndpoint or getAuthToken must
+     * be provided.
+     */
+    getAuthToken?: () => Promise<string>;
+    /**
+     * The user name for trusted auth
+     */
+    username?: string;
+
```
Comment (ashubham) [inline src/types.ts:66]: "password" for Basic auth.

**Review by ashubham — APPROVED** [review]

File: src/types.ts (hunk 1 above)
Comment (ayonghosh) [reply src/types.ts:27]: I'll send this out in a subsequent PR.

File: src/types.ts (hunk 2 above)
Comment (ayonghosh) [reply src/types.ts:66]: Will add this in a new PR after some thought around the security implications.

**Pull request merged**